e2core/server/server.go

```diff
@@ -31,51 +37,57 @@ func New(sync *syncer.Syncer, opts *options.Options) (*Server, error) {
 		return nil, errors.Wrapf(err, "setupTracing(%s, %s, %f)", "e2core", "reporter_uri", 0.04)
 	}
 
-	s := &Server{
-		coordinator: coordinator.New(sync, opts),
-		syncer:      sync,
-		options:     opts,
+	busOpts := []bus.OptionsModifier{
+		bus.UseLogger(opts.Logger()),
 	}
 
-	// set up the server so that Server can inspect
-	// each request to trigger Router re-generation
-	// when needed (during headless mode).
-	s.server = vk.New(
+	busOpts = append(busOpts, bus.UseMeshTransport(websocket.New()))
```

Comment:
I can do a deep dive into bus later but my immediate questions would be this: 

What happens in the case where a websocket connection needs to be reestablished but requests keep coming in?

Will the Bus queue the newly registered requests for sending once a connection has been acquired? (broken pipe scenario)

Will the Bus immediately fail any requests until it has an established connection?

Can/Does the Bus timeout stalled requests?

Do individual scheduler workers maintain their own job queue or do they pull from a global job(bus) queue one at a time? 

If they maintain their own job queue and the worker dies what happens to buffered tasks?

Reply:
note: feel free to consider these as future considerations if there isn't an answer for how these situations are handled; it's just good to understand how the system under certain circumstances.

Reply:
The bus will try to do the right thing, but will err on the side of returning an error to the caller if something goes wrong. The two main failure cases are "there's no sats to handle this" or "we tried and it timed out". Both of those cases are handled and result in a returned error.

The e2core server does client-side load balancing, so it will distribute jobs to sats using `bus.Tunnel`.

There's no persistence, as we're ok with being lossy in certain cases, but we'll likely add some persistence when we implement scale-to-zero.